sdc/rewrites/dataframe_constructor.py

```diff
@@ -0,0 +1,200 @@
+            old_columns = args['columns'] if 'columns' in args else None
+
+            args['data'], args['columns'] = extract(args, self._block, self._func_ir)
+
+            args['data'] = self._replace_with_arrays(args['data'], stmt, self._block, self._func_ir)
+
+            self._replace_call(stmt, init_df_stmt.target, args, self._block, self._func_ir)
+
+            cleanup(old_data, old_columns, self._block, self._func_ir)
+
+        return self._block
+
+
+    def _reset(self):
+        self._block = None
+        self._func_ir = None
+        self._calls_to_rewrite = None
+
+
+    @staticmethod
+    def _match_dict_case(args, block, func_ir):
+        if 'data' in args and is_dict(args['data'], block, func_ir):
+            return True
+
+        return False
+
+
+    @staticmethod
+    def _match_tuple_case(args, block, func_ir):
+        for name in ('data', 'columns'):
+            if name in args and is_tuple(args[name], block, func_ir):
+                return True
+
+        return False
+
+
+    @staticmethod
+    def _extract_tuple_args(args, block, func_ir):
+        data_args = get_tuple_items(args['data'], block, func_ir) if 'data' in args else None
+        columns_args = get_tuple_items(args['columns'], block, func_ir) if 'columns' in args else None
+
+        return data_args, columns_args
+
+
+    @staticmethod
+    def _extract_dict_args(args, block, func_ir):
+        dict_items = get_dict_items(args['data'], block, func_ir)
+
+        data_args = [item[1] for item in dict_items]
+        columns_args = [item[0] for item in dict_items]
+
+        return data_args, columns_args
+
+
+    @staticmethod
+    def _remove_dict_var(data, columns, block, func_ir):
+        assert columns is None
+
+        remove_unused(data, block, func_ir)
+
+
+    @staticmethod
+    def _replace_call(stmt, new_call, args, block, func_ir):
+        func = stmt.value
+
+        data_args = args['data']
+        columns_args = args['columns']
+
+        index_args = args['index'] if 'index' in args else None
+
+        if index_args is None:
+            none_stmt = declare_constant(None, block, func_ir, stmt.loc)
+            index_args = none_stmt.target
+
+        index_args = RewriteDataFrame._replace_with_arrays([index_args], stmt, block, func_ir)
+
+        all_args = data_args + index_args + columns_args
+
+        call = Expr.call(new_call, all_args, {}, func.loc)
+
+        stmt.value = call
+
+    @staticmethod
+    def _replace_with_arrays(args, stmt, block, func_ir):
+        new_args = []
+
+        if isinstance(args, list):
+            for var in args:
+                if is_list(var, block, func_ir):
+                    call_stmt = make_call(fix_df_array, [var], {}, block, func_ir, var.loc)
+                    insert_before(block, call_stmt, stmt)
+                    new_args.append(call_stmt.target)
+                else:
+                    new_args.append(var)
+
+            return new_args
+
+        raise ValueError()
```

Comment:
Do not we want to add message for the exception?

Reply:
Completely removed